objects/69ead2/contained/DUESETUP/script.lua

```diff
@@ -0,0 +1,643 @@
+end
+
+function onLoad(saved_data)
+    -- this just makes the letter "i" above this object, if visuals change in any way, this can be removed
+    self.createButton({
+        click_function = "null",
+        function_owner = self,
+        label = "i",
+        position = {0,0.3,0.1},
+        width = 0,
+        height = 0,
+        font_size = 450,
+        font_color = {1,1,1},
+    })
+
+    if saved_data ~= "" then
+        local loaded_data = JSON.decode(saved_data)
+        setupComplete = loaded_data.setupComplete
+        impendTable = loaded_data.impendTable
+        g3Selected = loaded_data.g3Selected
+    end
+
+    if setupComplete then
+        createPanelButtons()
+        createAllCardButtons()
+    end
+end
+
+function null()
+end
+
+function doSetup(params)
+    local color = params.color
+    local panel = Global.call("getSpirit", {name = spiritName})
+    self.locked = true
+    self.addTag("Modify Cost")
+    local position = panel.getPosition() + Vector(-8.2,-0.22,6.8)
+    self.setPosition(position)
+    
+    -- hand for impending cards:
+    local position1 = Player[color].getHandTransform(2).position
+    position1.z = position1.z - 5.5
+    Global.call("SpawnHand", {color = color, position = position1})
+    
+    Wait.time(function() createPanelButtons() end, 0.1) -- without a wait, if spirit chosen after setup, buttons don't appear
+    
+    return true
+end
+
+-----------------------------------------
+-- buttons and functions for spirit panel
+-----------------------------------------
+
+function createPanelButtons()
+    local dances = Global.call("getSpirit", {name = spiritName})
+    dances.createButton({
+        click_function = "impendCard",
+        function_owner = self,
+        label = "Impend a card!",
+        position = {-0.93,0.5,0},
+        scale = {x=0.08, y=0.08, z=0.08},
+        width = 5000,
+        height = 750,
+        font_size = 750,
+        color = {1,1,1},
+        font_color = {0,0,0},
+        tooltip = "Place a card on the panel art\nand press the button to impend it!",
+    })
+    
+    dances.createButton({
+        click_function = "g3Toggle",
+        function_owner = self,
+        label = "☐\n\n\n", 
+        position = {1.1,0.2,-0.78},
+        scale = {x=0.08, y=0.08, z=0.08},
+        width = 1800,
+        height = 1800,
+        font_size = 450,
+        color = {0,0,0,0.5},
+        font_color = {0,0,0,100},
+        tooltip = "Press this if you are choosing the third growth option in order to show buttons on impending cards.",
+    })
+    if g3Selected then
+        local ind = getButtonIndex(dances, "g3Toggle")
+        dances.editButton({
+            index = ind,
+            label = "☑\n\n\n",
+        })
+    end
+end
+
+function impendCard(obj)
+    local color = Global.call("getSpiritColor", {name = spiritName})
+    local hits = Physics.cast({
+        origin = obj.getPosition() + Vector(-5,0,2),
+        direction = Vector(0,1,0),
+        type = 3,
```

Comment:
Partially. I wanted it to be a region since the player may have a strategy that leads them into impending many cards in a turn and individually going back and forth pressing the button could be slow and annoying, the box cast can allow them to just drag out multiple cards directly from their hand and do it all at once. I figured that with it being a very infrequent press (a couple times at the start of a turn, maybe only once) that a box cast wouldn't end up particularly resource demanding.
It would be entirely possible to have there be something like a snap point and a ray cast or even a detection system to do this for individual cards, but I found this to aid myself and so hopefully it can benefit other players too.